src/modules/price-router/PriceRouter.sol

```diff
@@ -41,7 +61,109 @@ contract PriceRouter is Ownable, AutomationCompatibleInterface {
      */
     mapping(ERC20 => AssetSettings) public getAssetSettings;
 
-    // ======================================= ADAPTOR OPERATIONS =======================================
+    // ======================================= OWNERSHIP TRANSISITION =======================================
+
+    /**
+     * @notice Emitted when an ownership transition is started.
+     */
+    event OwnerTransitionStarted(address newOwner, uint256 startTime);
+
+    /**
+     * @notice Emitted when an ownership transition is cancelled.
+     */
+    event OwnerTransitionCancelled();
+
+    /**
+     * @notice Emitted when an ownership transition is completed.
+     */
+    event OwnerTransitionComplete(address newOwner);
+
+    /**
+     * @notice Attempted to call a function intended for Zero Id address.
+     */
+    error PriceRouter__OnlyCallableByZeroId();
+
+    /**
+     * @notice Attempted to transition owner to the zero address.
+     */
+    error PriceRouter__NewOwnerCanNotBeZero();
+
+    /**
+     * @notice Attempted to perform a restricted action while ownership transition is pending.
+     */
+    error PriceRouter__TransitionPending();
+
+    /**
+     * @notice Attempted to cancel or complete a transition when one is not active.
+     */
+    error PriceRouter__TransitionNotPending();
+
+    /**
+     * @notice Attempted to call `completeTransition` from an address that is not the pending owner.
+     */
+    error PriceRouter__OnlyCallableByPendingOwner();
+
+    /**
+     * @notice The amount of time it takes for an ownership transition to work.
+     */
+    uint256 public constant TRANSITION_PERIOD = 7 days;
```

Comment:
Why 7 days?

Reply:
So we want it to be long enough that governance can react if the multisig is acting maliciously, but short enough that if we see a chainlink feed is being deprecated, we can update our pricing info in time before our pricing calls start reverting, so 7 days is a nice middle ground.

Reply:
Right, the timeline is (soon):

- `editAsset` would occur, let's say it's malicious.
- Community sees it, and enacts a governance action which requires 2 days minimum. This could include arbitrary calls in the governance action itself, but it could be done after the vote has gone through if need be. 
- This gives enough time for it.